test/SecTester.Scan.Tests/DefaultScansTests.cs

```diff
@@ -0,0 +1,325 @@
+
+    // assert
+    await _commandDispatcher.Received(1)
+      .Execute(Arg.Is<DeleteScan>(x => x.Url.EndsWith($"/api/v1/scans/{ScanId}/delete") && x.Method == HttpMethod.Get));
+  }
+
+  [Fact]
+  public async Task GetScan_ReturnsScanState()
+  {
+    // arrange
+    var scanState = new ScanState(ScanStatus.Done);
+
+    _commandDispatcher.Execute(Arg.Any<GetScan>())
+      .Returns(Task.FromResult<ScanState?>(scanState));
+
+    // act
+    var result = await _sut.GetScan(ScanId);
+
+    // assert
+    result.Should().Be(scanState);
+    await _commandDispatcher.Received(1)
+      .Execute(Arg.Is<GetScan>(x =>
+        x.Url.EndsWith($"/api/v1/scans/{ScanId}") && x.Method == HttpMethod.Get && x.ExpectReply));
+  }
+
+  [Fact]
+  public async Task GetScan_WhenResultIsNull_ThrowError()
+  {
+    // arrange
+    _commandDispatcher.Execute(Arg.Any<GetScan>())
+      .Returns(Task.FromResult<ScanState?>(null));
+
+    // act 
+    var act = () => _sut.GetScan(ScanId);
+
+    // assert
+    await act.Should().ThrowAsync<SecTesterException>().WithMessage(NullResultMessage);
+  }
+
+  [Fact]
+  public async Task UploadHar_CreatesNewHar()
+  {
+    // arrange
+    var options = new UploadHarOptions(new Har(), "filename.har");
+
+    _commandDispatcher.Execute(Arg.Any<UploadHar>())
+      .Returns(Task.FromResult<Identifiable<string>?>(new Identifiable<string>(HarId)));
+
+    // act
+    var result = await _sut.UploadHar(options);
+
+    // assert
+    result.Should().Be(HarId);
+    await _commandDispatcher.Received(1)
+      .Execute(Arg.Is<UploadHar>(x =>
+        x.Url.EndsWith($"/api/v1/files") && x.Method == HttpMethod.Post && x.ExpectReply && (
+          x.Params == null || x.Params.All(y => y.Key != "discard"))));
+  }
+
+  [Fact]
+  public async Task UploadHar_WithDiscardOption_CreatesNewHar()
+  {
+    // arrange
+    var options = new UploadHarOptions(new Har(), "filename.har", true);
+
+    _commandDispatcher.Execute(Arg.Any<UploadHar>())
+      .Returns(Task.FromResult<Identifiable<string>?>(new Identifiable<string>(HarId)));
+
+    // act
+    var result = await _sut.UploadHar(options);
+
+    // assert
+    result.Should().Be(HarId);
+    await _commandDispatcher.Received(1)
+      .Execute(Arg.Is<UploadHar>(x =>
+        x.Url.EndsWith($"/api/v1/files") && x.Method == HttpMethod.Post && x.Params != null &&
+        x.Params.Any(y => y.Key == "discard" && y.Value == "true")));
+  }
+
+  [Fact]
+  public async Task UploadHar_PassesHarInfoInThePayload()
+  {
+    // arrange
+    var options = new UploadHarOptions(new Har(), "filename.har", true);
+
+    var jsonFileContent = _messageSerializer.SerializeJsonContent(options.Har);
+    var fileNameContent = new StringContent(options.FileName);
+
+    _commandDispatcher.Execute(Arg.Any<UploadHar>())
+      .Returns(Task.FromResult<Identifiable<string>?>(new Identifiable<string>(HarId)));
+
+    Func<HttpContent?, bool> uploadHarContentMatcher = httpContent =>
```

Comment:
Well, the expectation is too complex. You just need to make sure that one of an item of `MultipartFormDataContent` contains a field with a `jsonFileContent`